### PR TITLE
[Feature] Swagger to graphql source plugin

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Adds a new Gatsby source plugin that reads in a swagger file and produces GraphQL nodes for querying.

## Related Issue

None

## Motivation and Context

I could not find a swagger source plugin that provided the data we needed to properly create the pages we want.
This also gives us more control over how the GraphQL nodes are created and sourced.

## How Has This Been Tested?

Using a local instance already set up and pointed to an external doc project with a swagger file:

1. Install new dependencies: `yarn install`
1. Run tests: `yarn test`
1. Verify all Jest tests pass
1. Add a `SWAGGER_SOURCE_PATTERNS` environment variable to the development  and production `.env` files:

``` sh
SWAGGER_SOURCE_PATTERNS=**/petstore.yaml,**/petstore.json,**/swagger.json
```

5.  Run the development server
1. Verify development server successfully starts. (Some errors may be logged, but they are most likely not related to the changes in this PR).
1. Navigate to the graphQL playground
1. Verify the following types are listed on the left pane: `swaggerOpenApiInfo`,` swaggerOpenApiPath`, `swaggerOpenApiDefinition`
1. Create a graphql using these types
1. Verify data results reflects the data in the source files.
1. Stop the dev server
1. Run the build commands: `yarn build` or `yarn build:prod`
1. Verify static site is built successfully

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,7 @@
    GATSBY_SOURCE_TITLE=My Docs Site
    GATSBY_LAUNCH_SRC=//assets.adobedtm.com/<id>/launch-<id>.min.js  // url for your analytics script
    LOCAL_PROJECT_DIRECTORY=/absolute/path/to/docs/project
+   SWAGGER_SOURCE_PATTERNS=**/petstore.json,**/swagger.json
```
Comment (macdonst) [inline README.md:33]: Maybe we can prefix the variable with `GATSBY_` so ones used during the build match up?

File: package-lock.json [hunk 2]
```diff
@@ -128,6 +128,40 @@
       "resolved": "https://registry.npmjs.org/@adobe/react-spectrum-workflow/-/react-spectrum-workflow-1.0.1.tgz",
       "integrity": "sha512-5BM+oJyP4vjmqAV/HSJYLJzR3wDDbEOMUZEEVeKJ81cKyq0aJ6qF9A9tmsW6jb/73cE7PpLr6vb+xqBqTOYsTw=="
     },
```
Comment (macdonst) [inline package-lock.json:130]: We should really burn this file with fire. Let's all switch to yarn.

**Comment by jcalcaben** [comment]
@macdonst

1. Yes this PR just adds the plugin to the project. It doesn't do anything with the data at the moment. This is just to keep the PR small.
 
2. Yes, the swagger parser does support swagger v2.0 and openapi v3.0 specs according to their docs: https://apitools.dev/swagger-parser/docs/
   I don't think I've tested it with v3.0 specs though. I'll see about adding a test for that.

3. Yes, I don't see why not. The way I would see it working would be to fetch the data into a file in the cache and point the parser to that file.
But I would want to do that in a separate PR.

**Comment by macdonst** [comment]
@jcalcaben sounds good! We have lots more work ahead of us it seems.

**Pull request merged**